content/docs/reference/providers/pulumi/aws/s3/bucket.md

```diff
@@ -0,0 +1,1827 @@
+				kmsMasterKeyId: mykey.arn,
+				sseAlgorithm: "aws:kms",
+			},
+		},
+	},
+});
+```
+
+```python
+Coming soon
+```
+
+## Create a Bucket Resource
+
+{{< langchoose csharp >}}
+
+<div class="highlight"><pre class="chroma"><code class="language-typescript" data-lang="typescript"><span class="k">new</span> <span class="nx"><a href=/docs/reference/pkg/nodejs/pulumi/aws/s3/#Bucket>Bucket</a></span><span class="p">(</span><span class="nx">name</span>: <span class="kt"><a href=https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String>string</a></span><span class="p">,</span> <span class="nx">args?</span>: <span class="kt"><a href=/docs/reference/pkg/nodejs/pulumi/aws/s3/#BucketArgs>BucketArgs</a></span><span class="p">,</span> <span class="nx">opts?</span>: <span class="kt"><a href=/docs/reference/pkg/nodejs/pulumi/pulumi/#CustomResourceOptions>pulumi.CustomResourceOptions</a></span><span class="p">);</span></code></pre></div>
+
+```python
+def __init__(__self__, resource_name, opts=None, acceleration_status=None, acl=None, arn=None, bucket=None, bucket_prefix=None, cors_rules=None, force_destroy=None, hosted_zone_id=None, lifecycle_rules=None, loggings=None, object_lock_configuration=None, policy=None, region=None, replication_configuration=None, request_payer=None, server_side_encryption_configuration=None, tags=None, versioning=None, website=None, website_domain=None, website_endpoint=None, __props__=None)
+```
+
+```go
+func NewBucket(ctx *pulumi.Context, name string, args *BucketArgs, opts ...pulumi.ResourceOption) (*Bucket, error)
+
+```
+
+```csharp
+public Bucket(string name, BucketArgs? args = null, CustomResourceOptions? options = null)
+
+```
+
+Creates a Bucket resource with the given unique name, arguments, and options.
+
+{{% lang nodejs %}}
+<ul class="pl-10">
+    <li><strong>name</strong> &ndash; (Required) The unique name of the resulting resource.</li>
+    <li><strong>args</strong> &ndash; (Optional) The arguments to use to populate this resource's properties.</li>
+    <li><strong>opts</strong> &ndash; (Optional) A bag of options that control this resource's behavior.</li>
+</ul>
+{{% /lang %}}
+
+{{% lang go %}}
+<ul class="pl-10">
+    <li><strong>name</strong> &ndash; (Required) The unique name of the resulting resource.</li>
+    <li><strong>args</strong> &ndash; (Optional) The arguments to use to populate this resource's properties.</li>
+    <li><strong>opts</strong> &ndash; (Optional) A bag of options that control this resource's behavior.</li>
+</ul>
+{{% /lang %}}
+
+{{% lang csharp %}}
+<ul class="pl-10">
+    <li><strong>name</strong> &ndash; (Required) The unique name of the resulting resource.</li>
+    <li><strong>args</strong> &ndash; (Optional) The arguments to use to populate this resource's properties.</li>
+    <li><strong>opts</strong> &ndash; (Optional) A bag of options that control this resource's behavior.</li>
+</ul>
+{{% /lang %}}
+
+The following arguments are supported:
```

Comment:
I wish these tables could be more densely packed. For prose, I find our current layout relatively pleasant, although the limited horizontal width means that I often feel like our docs require a lot of scrolling. Should we consider using a more condensed layout and smaller font for the information like properties which should be more dense?

For instance, check out https://devcenter.heroku.com/articles/platform-api-reference#account-attributes.

Reply:
We chatted about this and I have some ideas.